src/0_utils.js

```diff
@@ -92,7 +92,12 @@ utils.cleanLinkData = function(linkData, config) {
 			linkData['data']['$desktop_url'] = linkData['data']['$desktop_url'].replace(/#r:[a-z0-9-_]+$/i, '').replace(/([\?\&]_branch_match_id=\d+)/, '');
 		}
 	}
-	linkData['data'] = goog.json.serialize(linkData['data'] || {});
+	try {
+		JSON.parse(linkData['data']);
+	}
+	catch (e) {
+		linkData['data'] = goog.json.serialize(linkData['data'] || {});
```

Comment:
hmm. wired use of bracket notation that we have going on here (I see above), someone must have been writing a lot of ruby ;). I prefer assigning without bracket notation when possible (i.e. `linkData.data`). cleaner / less characters to read. your call